### PR TITLE
refactor: remove remaining scala code

### PR DESCRIPTION
**Pull request by lenaschoenburg**
This rewrites all remaining scala files in Java to simplify our build process.

### CODE REVIEW

**Review by npepinpe — APPROVED** [review]
I'm from the wrong team, of course, but :rocket: 

Would you rather have a review from the ZPA team just in case? I'm also not a Scala expert, so I may have missed some subtle things.

File: zeebe/expression-language/src/main/java/io/camunda/zeebe/el/impl/FeelEvaluationResult.java
Comment (tmetzke) [inline zeebe/expression-language/src/main/java/io/camunda/zeebe/el/impl/FeelEvaluationResult.java:37]: ❓ Should this class be `public` like the others in this package? I think the Scala version of this was a public class (since it wasn't defined as private and in Scala things are public-by-default). Plus, we return instances of this class from the `FeelExpressionLanguage` (but only as the specialized type of an interface at runtime).

File: zeebe/expression-language/src/main/java/io/camunda/zeebe/el/impl/FeelVariableContext.java
Comment (tmetzke) [inline zeebe/expression-language/src/main/java/io/camunda/zeebe/el/impl/FeelVariableContext.java:17]: ❓ Similar to the `FeelEvaluationResult`, should this class be `public`?

File: zeebe/expression-language/src/main/java/io/camunda/zeebe/el/impl/FeelEvaluationResult.java
Comment (oleschoenburg) [reply zeebe/expression-language/src/main/java/io/camunda/zeebe/el/impl/FeelEvaluationResult.java:37]: I'd lean towards no because this class is an implementation detail and not part of our API. Is there any reason why we _should_ expose it as public (other than that it was public before)?

File: zeebe/expression-language/src/main/java/io/camunda/zeebe/el/impl/FeelVariableContext.java
Comment (oleschoenburg) [reply zeebe/expression-language/src/main/java/io/camunda/zeebe/el/impl/FeelVariableContext.java:17]: Same as above, as it's an implementation detail I'd say no.

File: zeebe/expression-language/src/main/java/io/camunda/zeebe/el/impl/FeelEvaluationResult.java
Comment (tmetzke) [reply zeebe/expression-language/src/main/java/io/camunda/zeebe/el/impl/FeelEvaluationResult.java:37]: I was wondering about the same thing. I can't think of any specific reason right now to say it _should_ to be public (since it's not part of our API) 👍

File: zeebe/expression-language/src/main/java/io/camunda/zeebe/el/impl/FeelEvaluationResult.java
Comment (tmetzke) [inline zeebe/expression-language/src/main/java/io/camunda/zeebe/el/impl/FeelEvaluationResult.java:164]: ❓ Do we have test cases that verify the behavior for lists of mixed types? My Scala knowledge is not proficient enough to derive it from the original code but my understanding is that the original code would skip non-`ValString` elements while here we only return a list if all elements are `ValString`s. But I might be wrong with that.

Comment (oleschoenburg) [reply zeebe/expression-language/src/main/java/io/camunda/zeebe/el/impl/FeelEvaluationResult.java:164]: This is also confirmed by the only usage of `getListOfStrings` which has this error message:
>  "Expected result of the expression '%s' to be 'ARRAY' containing 'STRING' items, but was 'ARRAY' containing at least one non-'STRING' item."

File: zeebe/feel/src/main/java/io/camunda/zeebe/feel/impl/FeelFunctionProvider.java
Comment (tmetzke) [inline zeebe/feel/src/main/java/io/camunda/zeebe/feel/impl/FeelFunctionProvider.java:30]: ❓ Last question from my end: do we anticipate this to become an issue? 😄 Do we know when this is called? Will this ever be called for our function provider? 😅

**Review by tmetzke — APPROVED** [review]
🚀 Great effort, @oleschoenburg 👍 
Not that I don't like Scala, but this helps streamline the tools and languages we use and separates concerns better 💯 
I'm also not a Scala expert, so my input here is also limited probably. I used a converter tool to generate Java code from the original Scala code to compare it to your Java code here. Couldn't spot any substantial differences (except for your code being more precise in some places, using newer JDK capabilities ⭐).

File: zeebe/feel/src/main/java/io/camunda/zeebe/feel/impl/FeelFunctionProvider.java
Comment (oleschoenburg) [reply zeebe/feel/src/main/java/io/camunda/zeebe/feel/impl/FeelFunctionProvider.java:30]: It's definitely a bit ugly but the way we override methods here means that this should never be called (until someone changes `JavaFunctionProvider` of course :sweat_smile:)

**Comment by oleschoenburg** [comment]
Had to rebase to get all of the new CI checks.

**Comment by tmetzke** [comment]
@oleschoenburg, let's see how many attempts we'll need for this. I am pretty sure someone is trying to block this from getting merged 😉

**Pull request merged**
